components/collapsible-panel/collapsible-panel.js

```diff
@@ -378,6 +386,10 @@ class CollapsiblePanel extends SkeletonMixin(FocusMixin(RtlMixin(LitElement))) {
 		if (changedProperties.has('noSticky')) {
 			this._stickyObserverUpdate();
 		}
+
+		if (changedProperties.has('type')) {
+			this._group?._updatePanelAttributes();
```

Comment:
And this handles the case where the subscription has already happened when `type` is either initialized or changes in the future.